src/operator/debounce.ts

```diff
@@ -46,21 +43,25 @@ class DebounceSubscriber<T, R> extends OuterSubscriber<T, R> {
 
   protected _next(value: T) {
     let subscription = this.durationSubscription;
-    const duration = tryCatch(this.durationSelector)(value);
+    let duration: Observable<number> | Promise<number> = null;
 
-    if (duration === errorObject) {
-      this.destination.error(errorObject.e);
-    } else {
-      this.value = value;
-      this.hasValue = true;
-      if (subscription) {
-        subscription.unsubscribe();
-        this.remove(subscription);
-      }
-      subscription = subscribeToResult(this, duration);
-      if (!subscription.isUnsubscribed) {
-        this.add(this.durationSubscription = subscription);
-      }
+    try {
+      duration = this.durationSelector.call(this, value);
+    } catch (err) {
+      this.destination.error(err);
+      return;
+    }
```

Comment:
I think @blesh's suggestion on [#1266 (comment)](https://github.com/ReactiveX/RxJS/pull/1266#discussion_r51300242) also applies here. (To move the try/catch block to a separate function.)